Building higher fails on bytecode only switches

/cc @pqwy

The package should not try to compile native code if opam's `ocaml-native` variable is false. Using `oasis` or `topkg` may solve the problem.

```
# opam-version 1.2.2
# os           darwin
# command      make
# path         /Users/dbuenzli/.opam/4.03.0+effects/build/higher.0.2
# compiler     4.03.0+effects
# exit-code    2
# env-file     /Users/dbuenzli/.opam/4.03.0+effects/build/higher.0.2/higher-58668-1872ec.env
# stdout-file  /Users/dbuenzli/.opam/4.03.0+effects/build/higher.0.2/higher-58668-1872ec.out
# stderr-file  /Users/dbuenzli/.opam/4.03.0+effects/build/higher.0.2/higher-58668-1872ec.err
### stdout ###
# ocamldep higher.mli > ._bcdi/higher.di
# ocamldep higher.ml > ._d/higher.d
# ocamlc -c higher.mli
# ocamlc -c higher.ml
# ocamlc -a                -o higher.cma higher.cmo
# ocamldep -native higher.mli > ._ncdi/higher.di
# ocamlopt -c higher.ml
### stderr ###
# File "higher.ml", line 1:
# Error: higher.cmi
# is not a compiled interface for this version of OCaml.
# It seems to be for a newer version of OCaml.
# make[1]: *** [higher.cmx] Error 2
# make: *** [native-code-library] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -8,12 +8,9 @@ RESULT=higher
 
 OCAMLFLAGS += $(shell ocamlc -bin-annot 2>/dev/null && echo -bin-annot)
 
-LIBINSTALL_FILES := higher.mli higher.cmi \
-                    higher.cma
-
+LIBINSTALL_FILES := higher.mli higher.cmi higher.cma
 LIBINSTALL_OPTIONAL_FILES := higher.cmt higher.cmti
-NATIVE_LIBINSTALL_FILES := \
-                    higher.cmx higher.cmxa higher.a
+NATIVE_LIBINSTALL_FILES := higher.cmx higher.cmxa higher.a
 
 all: byte-code-library native-code-library
 

--- a/Makefile
+++ b/Makefile
@@ -6,9 +6,12 @@ SOURCES=higher.mli higher.ml
 
 RESULT=higher
 
+OCAMLFLAGS += $(shell ocamlc -bin-annot 2>/dev/null && echo -bin-annot)
+
 LIBINSTALL_FILES := higher.mli higher.cmi \
                     higher.cma
 
+LIBINSTALL_OPTIONAL_FILES := higher.cmt higher.cmti
 NATIVE_LIBINSTALL_FILES := \
                     higher.cmx higher.cmxa higher.a
 
@@ -18,7 +21,8 @@ meta-install:
 	$(OCAMLFIND) install higher META
 
 byte-code-lib-install: byte-code-library
-	ocamlfind install -add higher $(LIBINSTALL_FILES)
+	ocamlfind install -add higher $(LIBINSTALL_FILES) \
+           -optional $(LIBINSTALL_OPTIONAL_FILES)
 
 examples-install:
 	ocamlfind install -add higher examples/*

--- a/Makefile
+++ b/Makefile
@@ -7,15 +7,29 @@ SOURCES=higher.mli higher.ml
 RESULT=higher
 
 LIBINSTALL_FILES := higher.mli higher.cmi \
-                    higher.cma \
+                    higher.cma
+
+NATIVE_LIBINSTALL_FILES := \
                     higher.cmx higher.cmxa higher.a
 
 all: byte-code-library native-code-library
 
+meta-install:
+	$(OCAMLFIND) install higher META
+
+byte-code-lib-install: byte-code-library
+	ocamlfind install -add higher $(LIBINSTALL_FILES)
+
 examples-install:
 	ocamlfind install -add higher examples/*
 
-install: libinstall examples-install
-uninstall: libuninstall
+native-lib-install: native-code-library
+	ocamlfind install -add higher $(NATIVE_LIBINSTALL_FILES)
+
+byte-code-install: meta-install byte-code-lib-install examples-install
+native-install: byte-code-install native-lib-install
+install: native-install
+uninstall:
+	$(OCAMLFIND) remove higher
 
 include $(OCAMLMAKEFILE)

--- a/opam
+++ b/opam
@@ -5,8 +5,14 @@ homepage: "https://github.com/ocamllabs/higher"
 bug-reports: "https://github.com/ocamllabs/higher/issues"
 dev-repo: "https://github.com/ocamllabs/higher.git"
 license: "MIT"
-build: [[make]]
-install: [[make "install"]]
+build: [
+  [make]        {ocaml-native}
+  [make "bcl"] {!ocaml-native}
+]
+install: [
+  [make "native-install"]     {ocaml-native}
+  [make "byte-code-install"] {!ocaml-native}
+]
 remove: [["ocamlfind" "remove" "higher"]]
 depends: [ "ocamlfind" ]
 tags: ["org:ocamllabs"]

